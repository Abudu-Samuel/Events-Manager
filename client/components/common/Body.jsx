import React from 'react';

const Body = () => (
  <div>
    <div id="intro" className="view hm-black-strong">
      <div className="container-fluid full-bg-img d-flex
       align-items-center justify-content-center">
        <div className="row d-flex land justify-content-center">
          <div className="text-center">
            <ul className="mb-5">
              <li>
                <h1 className="h2-responsive animated wow fadeInUp
                 font-weight-bold title white-text mr-5 mb-3">
                 Hey, wanna book an event center ?</h1>
                <h3 className="h3-responsive wow font-weight-bold animated
                 fadeInUp white-text mr-5 mb-4">Find your next experience</h3>
              </li>
              <li>
                <div className="row wow fadeIn mr-2" data-wow-delay="0.4s">
                  <div style={{ margin: '0px auto' }}>
                    <a href="#latest"><img className="img-fluid animated
                     bounce infinite" style={{ height: 50, bottom: -200 }} src="http://res.cloudinary.com/leumas/image/upload/v1527453763/zf9bspuveypekkdlhuzt.png" />
                    </a>
                  </div>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <main className="mt-3">
      <div className="container">
        <section id="latest" className="text-center">
          <h2 className="mb-3 font-weight-bold text-center">How it works</h2>
          <div className="row d-flex justify-content-center mb-4">
            <div className="col-md-4">
              <i className="fa fa-search fa-4x orange-text mb-2" />
              <h4 className="font-weight-bold">Search</h4>
            </div>
            <div className="col-md-4 mb-4">
              <i className="fa fa-bolt fa-4x indigo-text mb-2" />
              <h4 className="font-weight-bold">Availability</h4>
            </div>
            <div className="col-md-4 mb-4">
              <i className="fa fa-check fa-4x blue-text mb-2" />
              <h4 className="font-weight-bold">Book</h4>
            </div>
          </div>
        </section>
        <hr className="my-1" />
      </div>
    </main>
  </div>
);

export default Body;
