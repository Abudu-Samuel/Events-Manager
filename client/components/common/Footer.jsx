import React from 'react';

const Footer = () => (
  <div>
    <footer className="page-footer center-on-small-only default-color pt-0">
      <div className="footer-copyright">
        <div className="container-fluid">
                Â© 2017 Copyright: <a href="#"><strong> Events Manager.com
          </strong></a>
        </div>
      </div>
    </footer>
  </div>
);

export default Footer;
