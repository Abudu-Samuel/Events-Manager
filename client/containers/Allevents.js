import React from 'react';
import PopularCenter from '../containers/PopularCenter';
import TrendingCenters from '../containers/TrendingCenters';

/**
 * creates Navbar component
 *  @returns {funct} Navbar
 */
const Allevents = () => (
    <div className="space">
        <div className="container">
            <div className="row">
                <div className="col-md-12">
                    <h2 className="mb-2 font-weight-bold grey-text text-center">Events For You !</h2>
                </div>
            </div>
            <PopularCenter />
            <TrendingCenters />
        </div>
    </div>
);

export default Allevents;

