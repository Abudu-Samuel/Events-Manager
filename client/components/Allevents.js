import React from 'react';
import PopularCenter from '../components/PopularCenter';
import TrendingCenters from '../components/TrendingCenters';
import Navbar from '../components/Navbar';
import Footer from '../components/Footer';

/**
 * creates Navbar component
 *  @returns {funct} Navbar
 */
const Allevents = () => (
    <div className="space">
        <Navbar />
        <div className="container">
            <div className="row">
                <div className="col-md-12">
                    <h2 className="mb-2 font-weight-bold grey-text text-center">Events For You !</h2>
                </div>
            </div>
            <PopularCenter />
            <TrendingCenters />
        </div>
        <Footer />
    </div>
);

export default Allevents;

