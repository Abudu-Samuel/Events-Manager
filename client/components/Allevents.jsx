import React from 'react';
import { connect } from 'react-redux';
import PopularCenter from '../components/PopularCenter';
import TrendingCenters from '../components/TrendingCenters';
import Navbar from '../components/Navbar';
import Footer from '../components/Footer';
import * as userActions from '../actions/actionCreator';


/**
 * @description Allevents component
 *
 * @class Allevents
 *
 * @extends {React.Component}
 */
class Allevents extends React.Component {
  /**
   * Creates an instance of Allevents.
   *
   * @param {object} props
   *
   * @memberof Allevents
   */
  constructor(props) {
    super(props);
    this.state = {
      user: null,
      centers: [],
      events: [],
      fetchingCenters: false
    };
  }
  /**
   *@method componentWillMount
   *
   * @description React lifecycle hook
   *
   * @return {object} updated state
   *
   * @memberof Allevents
   */
  componentWillMount() {
    this.props.getAllCenters();
    this.props.getAllEvents();
  }

  /**
   * @method componentWillReceiveProps
   *
   * @description React lifecycle hook
   *
   * @param {object} nextProps
   *
   * @return {object} updated state
   *
   * @memberof Allevents
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.getCenters[0]) {
      this.setState({ centers: nextProps.getCenters, fetchingCenters: false });
    } else {
      this.setState({ fetchingCenters: true });
    }
    if (nextProps.getEvents) {
      this.setState({ events: nextProps.getEvents, fetchingEvents: false });
    } else {
      this.setState({ fetchingEvents: true });
    }
  }

  // getCenterId(event) {
  //   event.preventDefault();
  //   console.log(event.target.dataset.centerid);
  // }

  /**
   * @method render
   *
   * @description React method render
   *
   * @return {jsx} Jsx representation of the dom
   *
   * @memberof Allevents
   */
  render() {
    return (
      <div className="space">
        <Navbar />
        <div className="container">
          <div className="row">
            <div className="col-md-12">
              <h2 className="font-weight-bold grey-text text-center">Events For You !</h2>
            </div>
          </div>
          <PopularCenter
            events =
              {this.state.events}/>
          <TrendingCenters
            centers =
              {this.state.centers}
            /* getCenterId = {this.getCenterId} */

          />
        </div>
        <Footer />
      </div>
    );
  }
}

/**
 * @description Redux connect parameter - mapDispatchToProps
 *
 * @param {function} state
 *
 * @param {function} ownProps
 *
 * @return {object} mapped dispatch
 */
function mapStateToProps(state, ownProps) {
  return {
    getCenters: state.centers.foundCenters,
    getEvents: state.events.foundEvents
  };
}

/**
 * @description Redux connect parameter - mapDispatchToProps
 *
 * @param {function} dispatch
 *
 * @return {object} mapped dispatch
 */
function mapDispatchToProps(dispatch) {
  return {
    getAllCenters: centerData => dispatch(userActions.getAllCenters(centerData)),
    getAllEvents: eventData => dispatch(userActions.getAllEvents(eventData))
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Allevents);
