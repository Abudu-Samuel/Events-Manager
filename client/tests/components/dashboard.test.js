import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { Allevents, mapDispatchToProps } from '../../components/common/Allevents';
import centerMockedData from '../__mocks__/centerMockedData';
import eventMockedData from '../__mocks__/eventMockedData';

const props = {
  getAllCenters: () => Promise.resolve(),
  getAllEvents: () => Promise.resolve()
};

const eventPage = {
  nextCenterPage: {
    pageData: {
      selected: 1
    }
  }
};

describe('create Component', () => {
  it('SHOULD RENDER COMPONENT CORRECTLY', () => {
    const wrapper = shallow(<Allevents {...props}/>);
    wrapper.setProps({
      getCenters: {
        centers: centerMockedData.centerData
      }
    });
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.state().fetchingCenters).toEqual(false);
  });

  it('SHOULD RENDER REACT LIFECYCLE FOR getEvents IF STATEMENT', () => {
    const wrapper = shallow(<Allevents {...props} />);
    wrapper.setProps({
      getEvents: {
        events: eventMockedData.eventData
      }
    });
    expect(wrapper.exists()).toBe(true);
    wrapper.instance().eventPaginate(eventPage);
    wrapper.instance().centerPaginate(eventPage);
    expect(wrapper.state().fetchingEvents).toEqual(false);
  });

  it('should return an object of functions to be passed as props', () => {
    const dispatchFn = jest.fn();
    const propsObj = mapDispatchToProps(dispatchFn);

    expect(propsObj).toEqual({
      getAllCenters: expect.any(Function),
      getAllEvents: expect.any(Function)
    });

    propsObj.getAllCenters();
    propsObj.getAllEvents();

    expect(dispatchFn).toHaveBeenCalledTimes(2);
  });
});
