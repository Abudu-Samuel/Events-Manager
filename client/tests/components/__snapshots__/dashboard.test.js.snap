// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create Component SHOULD RENDER COMPONENT CORRECTLY 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Allevents
    getAllCenters={[Function]}
    getAllEvents={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Navbar) />,
        <div
          className="space"
        >
          <div
            className="container"
            style={
              Object {
                "marginTop": 99,
              }
            }
          >
            <h3
              className="font-weight-bold"
            >
              Welcome to Events Manager
            </h3>
            <div>
              <PopularEvents
                events={
                  Object {
                    "event": Array [],
                  }
                }
              />
              <PaginationBoxView
                activeClassName="active"
                breakClassName="page-link"
                breakLabel={
                  <a
                    href=""
                  >
                    ...
                  </a>
                }
                containerClassName="pagination pagination-lg custom-pagination"
                disableInitialCallback={false}
                disabledClassName="disabled"
                marginPagesDisplayed={3}
                nextClassName="page-item"
                nextLabel="Next"
                nextLinkClassName="page-link next"
                onPageChange={[Function]}
                pageClassName="page-item"
                pageCount={NaN}
                pageLinkClassName="page-link"
                pageRangeDisplayed={2}
                previousClassName="page-item"
                previousLabel="Previous"
                previousLinkClassName="page-link previous"
                subContainerClassName="pages pagination"
              />
            </div>
            <hr
              className="my-5"
            />
            <div>
              <TrendingCenter
                centers={
                  Object {
                    "center": Array [],
                  }
                }
              />
              <PaginationBoxView
                activeClassName="active"
                breakClassName="page-link"
                breakLabel={
                  <a
                    href=""
                  >
                    ...
                  </a>
                }
                containerClassName="pagination pagination-lg custom-pagination"
                disableInitialCallback={false}
                disabledClassName="disabled"
                marginPagesDisplayed={3}
                nextClassName="page-item"
                nextLabel="Next"
                nextLinkClassName="page-link"
                onPageChange={[Function]}
                pageClassName="page-item"
                pageCount={NaN}
                pageLinkClassName="page-link"
                pageRangeDisplayed={2}
                previousClassName="page-item"
                previousLabel="Previous"
                previousLinkClassName="page-link"
                subContainerClassName="pages pagination"
              />
            </div>
          </div>
          <Footer />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="container"
              style={
                Object {
                  "marginTop": 99,
                }
              }
            >
              <h3
                className="font-weight-bold"
              >
                Welcome to Events Manager
              </h3>
              <div>
                <PopularEvents
                  events={
                    Object {
                      "event": Array [],
                    }
                  }
                />
                <PaginationBoxView
                  activeClassName="active"
                  breakClassName="page-link"
                  breakLabel={
                    <a
                      href=""
                    >
                      ...
                    </a>
                  }
                  containerClassName="pagination pagination-lg custom-pagination"
                  disableInitialCallback={false}
                  disabledClassName="disabled"
                  marginPagesDisplayed={3}
                  nextClassName="page-item"
                  nextLabel="Next"
                  nextLinkClassName="page-link next"
                  onPageChange={[Function]}
                  pageClassName="page-item"
                  pageCount={NaN}
                  pageLinkClassName="page-link"
                  pageRangeDisplayed={2}
                  previousClassName="page-item"
                  previousLabel="Previous"
                  previousLinkClassName="page-link previous"
                  subContainerClassName="pages pagination"
                />
              </div>
              <hr
                className="my-5"
              />
              <div>
                <TrendingCenter
                  centers={
                    Object {
                      "center": Array [],
                    }
                  }
                />
                <PaginationBoxView
                  activeClassName="active"
                  breakClassName="page-link"
                  breakLabel={
                    <a
                      href=""
                    >
                      ...
                    </a>
                  }
                  containerClassName="pagination pagination-lg custom-pagination"
                  disableInitialCallback={false}
                  disabledClassName="disabled"
                  marginPagesDisplayed={3}
                  nextClassName="page-item"
                  nextLabel="Next"
                  nextLinkClassName="page-link"
                  onPageChange={[Function]}
                  pageClassName="page-item"
                  pageCount={NaN}
                  pageLinkClassName="page-link"
                  pageRangeDisplayed={2}
                  previousClassName="page-item"
                  previousLabel="Previous"
                  previousLinkClassName="page-link"
                  subContainerClassName="pages pagination"
                />
              </div>
            </div>,
            <Footer />,
          ],
          "className": "space",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="font-weight-bold"
                >
                  Welcome to Events Manager
                </h3>,
                <div>
                  <PopularEvents
                    events={
                      Object {
                        "event": Array [],
                      }
                    }
                  />
                  <PaginationBoxView
                    activeClassName="active"
                    breakClassName="page-link"
                    breakLabel={
                      <a
                        href=""
                      >
                        ...
                      </a>
                    }
                    containerClassName="pagination pagination-lg custom-pagination"
                    disableInitialCallback={false}
                    disabledClassName="disabled"
                    marginPagesDisplayed={3}
                    nextClassName="page-item"
                    nextLabel="Next"
                    nextLinkClassName="page-link next"
                    onPageChange={[Function]}
                    pageClassName="page-item"
                    pageCount={NaN}
                    pageLinkClassName="page-link"
                    pageRangeDisplayed={2}
                    previousClassName="page-item"
                    previousLabel="Previous"
                    previousLinkClassName="page-link previous"
                    subContainerClassName="pages pagination"
                  />
                </div>,
                <hr
                  className="my-5"
                />,
                <div>
                  <TrendingCenter
                    centers={
                      Object {
                        "center": Array [],
                      }
                    }
                  />
                  <PaginationBoxView
                    activeClassName="active"
                    breakClassName="page-link"
                    breakLabel={
                      <a
                        href=""
                      >
                        ...
                      </a>
                    }
                    containerClassName="pagination pagination-lg custom-pagination"
                    disableInitialCallback={false}
                    disabledClassName="disabled"
                    marginPagesDisplayed={3}
                    nextClassName="page-item"
                    nextLabel="Next"
                    nextLinkClassName="page-link"
                    onPageChange={[Function]}
                    pageClassName="page-item"
                    pageCount={NaN}
                    pageLinkClassName="page-link"
                    pageRangeDisplayed={2}
                    previousClassName="page-item"
                    previousLabel="Previous"
                    previousLinkClassName="page-link"
                    subContainerClassName="pages pagination"
                  />
                </div>,
              ],
              "className": "container",
              "style": Object {
                "marginTop": 99,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Welcome to Events Manager",
                  "className": "font-weight-bold",
                },
                "ref": null,
                "rendered": "Welcome to Events Manager",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <PopularEvents
                      events={
                        Object {
                          "event": Array [],
                        }
                      }
                    />,
                    <PaginationBoxView
                      activeClassName="active"
                      breakClassName="page-link"
                      breakLabel={
                        <a
                          href=""
                        >
                          ...
                        </a>
                      }
                      containerClassName="pagination pagination-lg custom-pagination"
                      disableInitialCallback={false}
                      disabledClassName="disabled"
                      marginPagesDisplayed={3}
                      nextClassName="page-item"
                      nextLabel="Next"
                      nextLinkClassName="page-link next"
                      onPageChange={[Function]}
                      pageClassName="page-item"
                      pageCount={NaN}
                      pageLinkClassName="page-link"
                      pageRangeDisplayed={2}
                      previousClassName="page-item"
                      previousLabel="Previous"
                      previousLinkClassName="page-link previous"
                      subContainerClassName="pages pagination"
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "events": Object {
                        "event": Array [],
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "activeClassName": "active",
                      "breakClassName": "page-link",
                      "breakLabel": <a
                        href=""
                      >
                        ...
                      </a>,
                      "containerClassName": "pagination pagination-lg custom-pagination",
                      "disableInitialCallback": false,
                      "disabledClassName": "disabled",
                      "marginPagesDisplayed": 3,
                      "nextClassName": "page-item",
                      "nextLabel": "Next",
                      "nextLinkClassName": "page-link next",
                      "onPageChange": [Function],
                      "pageClassName": "page-item",
                      "pageCount": NaN,
                      "pageLinkClassName": "page-link",
                      "pageRangeDisplayed": 2,
                      "previousClassName": "page-item",
                      "previousLabel": "Previous",
                      "previousLinkClassName": "page-link previous",
                      "subContainerClassName": "pages pagination",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "my-5",
                },
                "ref": null,
                "rendered": null,
                "type": "hr",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <TrendingCenter
                      centers={
                        Object {
                          "center": Array [],
                        }
                      }
                    />,
                    <PaginationBoxView
                      activeClassName="active"
                      breakClassName="page-link"
                      breakLabel={
                        <a
                          href=""
                        >
                          ...
                        </a>
                      }
                      containerClassName="pagination pagination-lg custom-pagination"
                      disableInitialCallback={false}
                      disabledClassName="disabled"
                      marginPagesDisplayed={3}
                      nextClassName="page-item"
                      nextLabel="Next"
                      nextLinkClassName="page-link"
                      onPageChange={[Function]}
                      pageClassName="page-item"
                      pageCount={NaN}
                      pageLinkClassName="page-link"
                      pageRangeDisplayed={2}
                      previousClassName="page-item"
                      previousLabel="Previous"
                      previousLinkClassName="page-link"
                      subContainerClassName="pages pagination"
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "centers": Object {
                        "center": Array [],
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "activeClassName": "active",
                      "breakClassName": "page-link",
                      "breakLabel": <a
                        href=""
                      >
                        ...
                      </a>,
                      "containerClassName": "pagination pagination-lg custom-pagination",
                      "disableInitialCallback": false,
                      "disabledClassName": "disabled",
                      "marginPagesDisplayed": 3,
                      "nextClassName": "page-item",
                      "nextLabel": "Next",
                      "nextLinkClassName": "page-link",
                      "onPageChange": [Function],
                      "pageClassName": "page-item",
                      "pageCount": NaN,
                      "pageLinkClassName": "page-link",
                      "pageRangeDisplayed": 2,
                      "previousClassName": "page-item",
                      "previousLabel": "Previous",
                      "previousLinkClassName": "page-link",
                      "subContainerClassName": "pages pagination",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Navbar) />,
          <div
            className="space"
          >
            <div
              className="container"
              style={
                Object {
                  "marginTop": 99,
                }
              }
            >
              <h3
                className="font-weight-bold"
              >
                Welcome to Events Manager
              </h3>
              <div>
                <PopularEvents
                  events={
                    Object {
                      "event": Array [],
                    }
                  }
                />
                <PaginationBoxView
                  activeClassName="active"
                  breakClassName="page-link"
                  breakLabel={
                    <a
                      href=""
                    >
                      ...
                    </a>
                  }
                  containerClassName="pagination pagination-lg custom-pagination"
                  disableInitialCallback={false}
                  disabledClassName="disabled"
                  marginPagesDisplayed={3}
                  nextClassName="page-item"
                  nextLabel="Next"
                  nextLinkClassName="page-link next"
                  onPageChange={[Function]}
                  pageClassName="page-item"
                  pageCount={NaN}
                  pageLinkClassName="page-link"
                  pageRangeDisplayed={2}
                  previousClassName="page-item"
                  previousLabel="Previous"
                  previousLinkClassName="page-link previous"
                  subContainerClassName="pages pagination"
                />
              </div>
              <hr
                className="my-5"
              />
              <div>
                <TrendingCenter
                  centers={
                    Object {
                      "center": Array [],
                    }
                  }
                />
                <PaginationBoxView
                  activeClassName="active"
                  breakClassName="page-link"
                  breakLabel={
                    <a
                      href=""
                    >
                      ...
                    </a>
                  }
                  containerClassName="pagination pagination-lg custom-pagination"
                  disableInitialCallback={false}
                  disabledClassName="disabled"
                  marginPagesDisplayed={3}
                  nextClassName="page-item"
                  nextLabel="Next"
                  nextLinkClassName="page-link"
                  onPageChange={[Function]}
                  pageClassName="page-item"
                  pageCount={NaN}
                  pageLinkClassName="page-link"
                  pageRangeDisplayed={2}
                  previousClassName="page-item"
                  previousLabel="Previous"
                  previousLinkClassName="page-link"
                  subContainerClassName="pages pagination"
                />
              </div>
            </div>
            <Footer />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="container"
                style={
                  Object {
                    "marginTop": 99,
                  }
                }
              >
                <h3
                  className="font-weight-bold"
                >
                  Welcome to Events Manager
                </h3>
                <div>
                  <PopularEvents
                    events={
                      Object {
                        "event": Array [],
                      }
                    }
                  />
                  <PaginationBoxView
                    activeClassName="active"
                    breakClassName="page-link"
                    breakLabel={
                      <a
                        href=""
                      >
                        ...
                      </a>
                    }
                    containerClassName="pagination pagination-lg custom-pagination"
                    disableInitialCallback={false}
                    disabledClassName="disabled"
                    marginPagesDisplayed={3}
                    nextClassName="page-item"
                    nextLabel="Next"
                    nextLinkClassName="page-link next"
                    onPageChange={[Function]}
                    pageClassName="page-item"
                    pageCount={NaN}
                    pageLinkClassName="page-link"
                    pageRangeDisplayed={2}
                    previousClassName="page-item"
                    previousLabel="Previous"
                    previousLinkClassName="page-link previous"
                    subContainerClassName="pages pagination"
                  />
                </div>
                <hr
                  className="my-5"
                />
                <div>
                  <TrendingCenter
                    centers={
                      Object {
                        "center": Array [],
                      }
                    }
                  />
                  <PaginationBoxView
                    activeClassName="active"
                    breakClassName="page-link"
                    breakLabel={
                      <a
                        href=""
                      >
                        ...
                      </a>
                    }
                    containerClassName="pagination pagination-lg custom-pagination"
                    disableInitialCallback={false}
                    disabledClassName="disabled"
                    marginPagesDisplayed={3}
                    nextClassName="page-item"
                    nextLabel="Next"
                    nextLinkClassName="page-link"
                    onPageChange={[Function]}
                    pageClassName="page-item"
                    pageCount={NaN}
                    pageLinkClassName="page-link"
                    pageRangeDisplayed={2}
                    previousClassName="page-item"
                    previousLabel="Previous"
                    previousLinkClassName="page-link"
                    subContainerClassName="pages pagination"
                  />
                </div>
              </div>,
              <Footer />,
            ],
            "className": "space",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3
                    className="font-weight-bold"
                  >
                    Welcome to Events Manager
                  </h3>,
                  <div>
                    <PopularEvents
                      events={
                        Object {
                          "event": Array [],
                        }
                      }
                    />
                    <PaginationBoxView
                      activeClassName="active"
                      breakClassName="page-link"
                      breakLabel={
                        <a
                          href=""
                        >
                          ...
                        </a>
                      }
                      containerClassName="pagination pagination-lg custom-pagination"
                      disableInitialCallback={false}
                      disabledClassName="disabled"
                      marginPagesDisplayed={3}
                      nextClassName="page-item"
                      nextLabel="Next"
                      nextLinkClassName="page-link next"
                      onPageChange={[Function]}
                      pageClassName="page-item"
                      pageCount={NaN}
                      pageLinkClassName="page-link"
                      pageRangeDisplayed={2}
                      previousClassName="page-item"
                      previousLabel="Previous"
                      previousLinkClassName="page-link previous"
                      subContainerClassName="pages pagination"
                    />
                  </div>,
                  <hr
                    className="my-5"
                  />,
                  <div>
                    <TrendingCenter
                      centers={
                        Object {
                          "center": Array [],
                        }
                      }
                    />
                    <PaginationBoxView
                      activeClassName="active"
                      breakClassName="page-link"
                      breakLabel={
                        <a
                          href=""
                        >
                          ...
                        </a>
                      }
                      containerClassName="pagination pagination-lg custom-pagination"
                      disableInitialCallback={false}
                      disabledClassName="disabled"
                      marginPagesDisplayed={3}
                      nextClassName="page-item"
                      nextLabel="Next"
                      nextLinkClassName="page-link"
                      onPageChange={[Function]}
                      pageClassName="page-item"
                      pageCount={NaN}
                      pageLinkClassName="page-link"
                      pageRangeDisplayed={2}
                      previousClassName="page-item"
                      previousLabel="Previous"
                      previousLinkClassName="page-link"
                      subContainerClassName="pages pagination"
                    />
                  </div>,
                ],
                "className": "container",
                "style": Object {
                  "marginTop": 99,
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Welcome to Events Manager",
                    "className": "font-weight-bold",
                  },
                  "ref": null,
                  "rendered": "Welcome to Events Manager",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <PopularEvents
                        events={
                          Object {
                            "event": Array [],
                          }
                        }
                      />,
                      <PaginationBoxView
                        activeClassName="active"
                        breakClassName="page-link"
                        breakLabel={
                          <a
                            href=""
                          >
                            ...
                          </a>
                        }
                        containerClassName="pagination pagination-lg custom-pagination"
                        disableInitialCallback={false}
                        disabledClassName="disabled"
                        marginPagesDisplayed={3}
                        nextClassName="page-item"
                        nextLabel="Next"
                        nextLinkClassName="page-link next"
                        onPageChange={[Function]}
                        pageClassName="page-item"
                        pageCount={NaN}
                        pageLinkClassName="page-link"
                        pageRangeDisplayed={2}
                        previousClassName="page-item"
                        previousLabel="Previous"
                        previousLinkClassName="page-link previous"
                        subContainerClassName="pages pagination"
                      />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "events": Object {
                          "event": Array [],
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "activeClassName": "active",
                        "breakClassName": "page-link",
                        "breakLabel": <a
                          href=""
                        >
                          ...
                        </a>,
                        "containerClassName": "pagination pagination-lg custom-pagination",
                        "disableInitialCallback": false,
                        "disabledClassName": "disabled",
                        "marginPagesDisplayed": 3,
                        "nextClassName": "page-item",
                        "nextLabel": "Next",
                        "nextLinkClassName": "page-link next",
                        "onPageChange": [Function],
                        "pageClassName": "page-item",
                        "pageCount": NaN,
                        "pageLinkClassName": "page-link",
                        "pageRangeDisplayed": 2,
                        "previousClassName": "page-item",
                        "previousLabel": "Previous",
                        "previousLinkClassName": "page-link previous",
                        "subContainerClassName": "pages pagination",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "my-5",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "hr",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <TrendingCenter
                        centers={
                          Object {
                            "center": Array [],
                          }
                        }
                      />,
                      <PaginationBoxView
                        activeClassName="active"
                        breakClassName="page-link"
                        breakLabel={
                          <a
                            href=""
                          >
                            ...
                          </a>
                        }
                        containerClassName="pagination pagination-lg custom-pagination"
                        disableInitialCallback={false}
                        disabledClassName="disabled"
                        marginPagesDisplayed={3}
                        nextClassName="page-item"
                        nextLabel="Next"
                        nextLinkClassName="page-link"
                        onPageChange={[Function]}
                        pageClassName="page-item"
                        pageCount={NaN}
                        pageLinkClassName="page-link"
                        pageRangeDisplayed={2}
                        previousClassName="page-item"
                        previousLabel="Previous"
                        previousLinkClassName="page-link"
                        subContainerClassName="pages pagination"
                      />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "centers": Object {
                          "center": Array [],
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "activeClassName": "active",
                        "breakClassName": "page-link",
                        "breakLabel": <a
                          href=""
                        >
                          ...
                        </a>,
                        "containerClassName": "pagination pagination-lg custom-pagination",
                        "disableInitialCallback": false,
                        "disabledClassName": "disabled",
                        "marginPagesDisplayed": 3,
                        "nextClassName": "page-item",
                        "nextLabel": "Next",
                        "nextLinkClassName": "page-link",
                        "onPageChange": [Function],
                        "pageClassName": "page-item",
                        "pageCount": NaN,
                        "pageLinkClassName": "page-link",
                        "pageRangeDisplayed": 2,
                        "previousClassName": "page-item",
                        "previousLabel": "Previous",
                        "previousLinkClassName": "page-link",
                        "subContainerClassName": "pages pagination",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
